Assets/SEE/Game/City/SEECity.cs

```diff
@@ -345,30 +289,36 @@ public virtual void LoadAndDrawGraph()
         ///
         /// This method loads only the data, but does not actually render the graph.
         /// </summary>
-        [Button(ButtonSizes.Small)]
+        [Button(ButtonSizes.Small, Name = "Load Data")]
         [ButtonGroup(DataButtonsGroup), RuntimeButton(DataButtonsGroup, "Load Data")]
         [PropertyOrder(DataButtonsGroupOrderLoad)]
-        public virtual void LoadData()
+        public virtual async UniTask LoadDataAsync()
         {
-            if (string.IsNullOrEmpty(GXLPath.Path))
+            if (DataProvider != null)
             {
-                Debug.LogError("Empty graph path.\n");
+                try
+                {
+                    LoadedGraph = await DataProvider.ProvideAsync(new Graph(""), this);
+                }
+                catch (Exception ex)
+                {
+                    ShowNotification.Error("Data failure", $"Graph provider failed with: {ex}\n");
+                    Debug.LogException(ex);
+                }
             }
             else
             {
-                if (LoadedGraph != null)
-                {
-                    Reset();
-                }
-
-                LoadedGraph = LoadGraph(GXLPath.Path);
-                LoadMetrics();
+                ShowNotification.Error("No data provider", "You must set a data provider before you can load the data.");
             }
         }
 
         /// <summary>
         /// Saves the graph data to the GXL file with GXLPath().
         /// </summary>
+        /*
+         * FIXME: Do we want to re-active this code? If so, it should query the path from
```

Comment:
The CI didn't catch this FIXME due to the `*` not being recognized as a comment character, but it should still either be resolved or converted to an issue.

Reply:
I re-activated it. The graph is saved into a GXL file with a default name. That should do for the moment.